yt_dlp/extractor/cbc.py

```diff
@@ -228,6 +229,63 @@ def _real_extract(self, url):
         }
 
 
+class CBCPlayerPlaylistIE(InfoExtractor):
+    IE_NAME = 'cbc.ca:player:playlist'
+    _VALID_URL = r'(?:cbcplayer:|https?://(?:www\.)?cbc\.ca/(?:player/)(?!play/))(?P<id>.+)'
+    _TESTS = [{
+        'url': 'https://www.cbc.ca/player/news/TV%20Shows/The%20National/Latest%20Broadcast',
+        'playlist_mincount': 25,
+        'info_dict': {
+            'id': 'news/tv shows/the national/latest broadcast',
+            'ie_key': 'CBCPlayer',
+        }
+    }, {
+        'url': 'https://www.cbc.ca/player/news/Canada/North',
+        'playlist_mincount': 25,
+        'info_dict': {
+            'id': 'news/canada/north',
+            'ie_key': 'CBCPlayer',
+        }
+    }]
+
+    def _real_extract(self, url):
+        # We have no other playlist id other than the URL, so use that
+        playlist_id = self._match_id(url).replace('%20', ' ').lower()
+        # the json info we're looking for isn't marked as such, therefore we need a bit of a workaround.
+        json_content = self._parse_json(
+            re.sub(
+                r'^.*?{', '{',
+                get_element_by_id('initialStateDom', self._download_webpage(url, playlist_id))
+            )[:-1],
+            playlist_id)
+        playlist_items = []
+        for key, value in json_content.get('video').get('clipsByCategory').items():
+            # We need to do a case insensitive match. If anyone has a better way, feel free to improve.
+            if key.lower() == playlist_id:
+                for video in value.get('items'):
+                    playlist_items.append({
+                        '_type': 'url',
+                        'url': 'https://www.cbc.ca/player/play/%s' % video['id'],
+                        'ie_key': 'CBCPlayer',
+                        'timestamp': video['airDate'],
+                        'categories': [video['contentArea']],
+                        'description': video['description'],
+                        'duration': video['duration'],
+                        'id': video['id'],
+                        'is_live': video['isLive'],
+                        'series': video['showName'],
+                        'thumbnail': video['thumbnail'],
+                        'title': video['title'],
+                    })
+                break
+        return {
+            '_type': 'playlist',
+            'id': playlist_id,
+            'ie_key': 'CBCPlayer',
+            'entries': playlist_items,
+        }
```

Comment:
- Do not do `a.get('b').get('c')`, use `traverse_obj(a, ('b', 'c'))`
- Use `traverse_obj` with for loop, it will be non-fatal and it can be used to filter
- Metadata extraction should be non-fatal (not risking a KeyError), again use `traverse_obj` here
- Use `int_or_none` or `float_or_none` with metadata fields that need to be int/float
- video `id` should be string, use `str_or_none` (import the `or_none` helpers from utils)
- f-string is preferred for `url`
- Use a generator func instead of append loop

```suggestion
        def entries():
            for video in traverse_obj(json_content, (
                'video', 'clipsByCategory', lambda k, _: k.lower() == playlist_id, 'items', lambda _, v: v['id']
            )):
                yield self.url_result(
                    f'https://www.cbc.ca/player/play/{video["id"]}', CBCPlayerIE,
                    **traverse_obj(video, {
                        'id': ('id', {str_or_none}),
                        'title': 'title',
                        'description': 'description',
                        'series': 'showName',
                        'categories': ('contentArea', {lambda x: [x] if x else None}),
                        'timestamp': ('airDate', {int_or_none}),
                        'duration': ('duration', {int_or_none}),
                        'is_live': ('isLive', {bool}),
                        'thumbnail': ('thumbnail', {url_or_none}),
                    }))

        return self.playlist_result(entries(), playlist_id)
```

Reply:
Why don't we need IE Key here? Don't we want to tell yt-dlp to use the CBCPlayerIE so that it doesn't have to go hunting again?
Fine with the rest - at least as far as I can tell.

Reply:
You want the video extractor's `ie_key` in the ***`url_result`***, which is why it's the 2nd arg. You don't want it to be in the playlist result, since it is the result of the playlist extractor

Reply:
Ah OK... I'll be completely honest, Python definitely isn't my best language, I missed the ie_key... as long as it works optimally, I'm fine with it.